src/terminal.ts

```diff
@@ -166,80 +166,160 @@ export class FlatpakManifest {
     const module = this.module()
     const configOpts = (module['config-opts'] || []).join(' ')
 
-    let commands: Command[] = []
     switch (module.buildsystem) {
+      case undefined:
+      case 'autotools':
+        throw new Error('Autotools is not implemented yet')
+      case 'cmake':
+      case 'cmake-ninja':
+        return this.getCmakeCommands(rebuild, buildArgs, configOpts)
       case 'meson':
-        {
-          const mesonBuildDir = '_build'
-          buildArgs.push(`--filesystem=${this.workspace}/${mesonBuildDir}`)
-          if (!rebuild) {
-            commands.push(
-              new Command(
-                'flatpak',
-                [
-                  'build',
-                  ...buildArgs,
-                  this.repoDir,
-                  'meson',
-                  '--prefix',
-                  '/app',
-                  mesonBuildDir,
-                  configOpts,
-                ],
-                this.workspace,
-                this.isSandboxed
-              )
-            )
-          }
-          commands.push(
-            new Command(
-              'flatpak',
-              [
-                'build',
-                ...buildArgs,
-                this.repoDir,
-                'ninja',
-                '-C',
-                mesonBuildDir,
-              ],
-              this.workspace,
-              this.isSandboxed
-            )
-          )
-          commands.push(
-            new Command(
-              'flatpak',
-              [
-                'build',
-                ...buildArgs,
-                this.repoDir,
-                'meson',
-                'install',
-                '-C',
-                mesonBuildDir,
-              ],
-              this.workspace,
-              this.isSandboxed
-            )
-          )
-        }
-        break
+        return this.getMesonCommands(rebuild, buildArgs, configOpts)
       case 'simple':
-        {
-          commands = module['build-commands'].map((command) => {
-            return new Command(
-              'flatpak',
-              ['build', ...buildArgs, this.repoDir, command],
-              this.workspace,
-              this.isSandboxed
-            )
-          })
-        }
-        break
+        return this.getSimpleCommands(module['build-commands'], buildArgs)
+      case 'qmake':
+        throw new Error('Qmake is not implemented yet')
+    }
+    throw new Error('Failed to build application')
+  }
+  getCmakeCommands(rebuild: boolean, buildArgs: string[], configOpts: string): Command[] {
```

Comment:
Also nitpick: could use an empty line between the two functions

Reply:
Done